dvc/commands/data.py

```diff
@@ -0,0 +1,178 @@
+            ui.write(header)
+            if hint := cls.HINTS.get(stage):
+                ui.write(f"  ({hint})")
+
+            if isinstance(stage_status, dict):
+                items = [
+                    ": ".join([state, file])
+                    for file, state in stage_status.items()
+                ]
+            else:
+                items = stage_status
+
+            for item in items:
+                ui.write(f"\t[{color}]{item}[/]".expandtabs(8), styled=True)
+
+        if (hint := cls.HINTS.get("git_dirty")) and git_info.get("is_dirty"):
+            message = hint.format("other " if result else "")
+            ui.write(f"[blue]({message})[/]", styled=True)
+        return 0
+
+    def run(self) -> int:
+        with log_durations(logger.trace, "in data_status"):  # type: ignore
+            status = self.repo.data_status(
+                granular=self.args.granular,
+                untracked_files=self.args.untracked_files,
+                with_dirs=self.args.with_dirs,
+            )
+
+        if not self.args.unchanged:
+            status.pop("unchanged")  # type: ignore[misc]
+        if self.args.untracked_files == "no":
+            status.pop("untracked")
+        if self.args.json:
+            status.pop("git")  # type: ignore[misc]
+            ui.write_json(compact(status))
+            return 0
+        return self._show_status(status)
+
+
+def add_parser(subparsers, parent_parser):
+    data_parser = subparsers.add_parser(
+        "data",
+        parents=[parent_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    data_subparsers = data_parser.add_subparsers(
+        dest="cmd",
+        help="Use `dvc data CMD --help` to display command-specific help.",
+    )
+    fix_subparsers(data_subparsers)
+
+    DATA_STATUS_HELP = (
+        "Show changes between the last git commit, "
+        "the dvcfiles and the workspace."
+    )
+    data_status_parser = data_subparsers.add_parser(
+        "status",
+        parents=[parent_parser],
+        description=append_doc_link(DATA_STATUS_HELP, "data/status"),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        help=DATA_STATUS_HELP,
+    )
+    data_status_parser.add_argument(
+        "--json",
+        action="store_true",
+        default=False,
+        help="Show output in JSON format.",
+    )
+    data_status_parser.add_argument(
+        "--show-json",
+        action="store_true",
+        default=False,
+        dest="json",
+        help=argparse.SUPPRESS,
+    )
+    data_status_parser.add_argument(
+        "--granular",
+        action="store_true",
+        default=False,
+        help="Show granular file-level info for DVC-tracked directories.",
+    )
+    data_status_parser.add_argument(
+        "--unchanged",
+        action="store_true",
+        default=False,
+        help="Show unmodified DVC-tracked files.",
+    )
+    data_status_parser.add_argument(
+        "--untracked-files",
```

Comment:
One more...

Should the 2nd option be just `--untracked`? Unclear why `-files` is only in that name (plus it can be directories too).

Reply:
yep, let's introduce this alternative now, and deprecate the previous option, drop it later

Reply:
Git uses `--untracked-files`, so that's a major reason why this is this way. Also, `--untracked-files` at the moment is recursive, so the `files` is more correct. There are some questions regarding it's behaviour: https://github.com/iterative/dvc/issues/8061, so I'd defer it until then.

Reply:
OK but remember that Git's UI is not very good. We should not aim to copy it just because it's Git. "Untracked" can refer to plural files so both are equally correct, I think.